Nasal/physics.nas

```diff
@@ -30,24 +30,87 @@ var killengine = func
         setprop("/engines/active-engine/crash-engine", 1);
 }
 
-# Check if on water
-var poll_surface = func
+# Hydro system loop
+
+var anchor_pos = geo.Coord.new();
+var aircraft_pos = geo.Coord.new();
+var anchor_dist = 0.0;
+var rope_length = 0.0;
+var heading_diff = 0.0;
+var wind_from_heading  = 0.0;
+var aircraft_heading  = 0.0;
+var ground_speed = 0.0;
+var wind_speed = 0.0;
+
+var poll_hydro = func
 {
     var engine_rpm = getprop("fdm/jsbsim/propulsion/engine/engine-rpm");
     var hydro_active_norm = getprop("/fdm/jsbsim/hydro/active-norm");
     var ground_splash_norm = getprop("/environment/aircraft-effects/ground-splash-norm");
+    ground_speed = getprop("/velocities/groundspeed-kt");
 
     # Use engine RPM and speed to control ground splash if on the water
     # and below 2 meter AGL
     if (getprop("position/altitude-agl-m") < 2 and hydro_active_norm > 0) {
-        var groundspeed_half_pc = 0.005 * getprop("velocities/groundspeed-kt");
+        var groundspeed_half_pc = 0.005 * ground_speed;
         var engine_rpm_almost_nothing = 0.005 * 0.065 * engine_rpm;
 
         var splash_norm = std.max(engine_rpm_almost_nothing, groundspeed_half_pc);
         setprop("/environment/aircraft-effects/ground-splash-norm", splash_norm);
     }
     elsif (ground_splash_norm > 0)
         setprop("/environment/aircraft-effects/ground-splash-norm", ground_splash_norm - 0.005);
+
+    if (getprop("/controls/mooring/anchor") and !getprop("fdm/jsbsim/mooring/mooring-connected")) {
+        if (getprop("/fdm/jsbsim/mooring/anchor-length") == 0) {
+            wind_from_heading = getprop("/environment/wind-from-heading-deg");
+            #wind_from_heading = getprop("/local-weather/METAR/wind-direction-deg");
+            aircraft_heading = getprop("/orientation/heading-deg");
+            heading_diff = math.min(math.abs(aircraft_heading-wind_from_heading),math.abs(360-aircraft_heading+wind_from_heading),math.abs(360-wind_from_heading+aircraft_heading));
+            wind_speed = getprop("/environment/windsock/wind-speed-kt");
+            if (heading_diff > 45) {
+                gui.popupTip("Face into the wind to set anchor", 5);
+                setprop("/controls/mooring/anchor", 0);
+                setprop("Aa-heading-diff", heading_diff);
+                return;
+            } else
```

Comment:
The formatting of the code here is weird. You can use `elsif` instead of `else if`, so you get:

```javascript
if (a) {
    //
}
elsif (b) {
    //
}
elsif (c) {
    //
}
else {
    //
}
```